routers/web/repo/issue_content_history.go

```diff
@@ -93,9 +93,15 @@ func canSoftDeleteContentHistory(ctx *context.Context, issue *issues_model.Issue
 	history *issues_model.ContentHistory,
 ) bool {
 	canSoftDelete := false
-	if ctx.Repo.IsOwner() {
+	// CanWrite means the doer can manage the issue list
+	if ctx.Repo.IsOwner() || ctx.Repo.CanWrite(unit.TypeIssues) {
```

Comment:
If issue is pull request, we should check `unit.TypePullRequest`.

Reply:
Good catch. Done in f10db1ffc23f68dba30beb6c99b33d3abad8e478